src/main/java/org/motechproject/newebodac/repository/custom/impl/FieldConfigRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package org.motechproject.newebodac.repository.custom.impl;
+
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.ParameterExpression;
+import javax.persistence.criteria.Root;
+
+import org.motechproject.newebodac.domain.FieldConfig;
+import org.motechproject.newebodac.domain.enums.EntityType;
+import org.motechproject.newebodac.dto.BaseDto;
+import org.motechproject.newebodac.exception.NewEbodacException;
+import org.motechproject.newebodac.repository.custom.FieldConfigRepositoryCustom;
+import org.motechproject.newebodac.repository.FieldConfigRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class FieldConfigRepositoryImpl extends BaseRepositoryImpl implements FieldConfigRepositoryCustom {
+  /**
+   * Finds entities matching all of the provided parameters. If there are no
```

Comment:
like I show you before it will be better to declare it like that (the EntityManager and CriteriaBuilder):
`
  private final EntityManager em;
  private final CriteriaBuilder cb;

  public FieldConfigRepositoryImpl(EntityManager em) {
    this.em = em;
    this.cb = em.getCriteriaBuilder();
  }
`